packages/cipherstash-proxy-integration/src/multitenant/set_keyset_name.rs

```diff
@@ -0,0 +1,374 @@
+
+        // KEYSET_NAME IS SCOPED TO A CONNECTION
+        // The same client/connection is used for tests
+        let client = connect_with_tls(PROXY).await;
+
+        // DEFAULT_KEYSET_ID SHOULD BE DISABLED FOR THIS TEST
+        // SET KEYSET IS REQUIRED
+        let result = client.query(select_sql, &[&42]).await;
+        assert!(result.is_err());
+
+        //  --------
+        // INSERT and SELECT as TENANT 1
+        // SET tenant 1
+        let sql = format!("SET CIPHERSTASH.KEYSET_NAME = '{tenant_keyset_name_1}'");
+        let result = client.query(&sql, &[]).await;
+        assert!(result.is_ok(), "Setting keyset_name should succeed");
+
+        // INSERT
+        let tenant_1_id = random_id();
+        let encrypted_val = 2;
+
+        client
+            .query(insert_sql, &[&tenant_1_id, &encrypted_val])
+            .await
+            .unwrap();
+
+        // SELECT
+        let rows = client.query(select_sql, &[&tenant_1_id]).await.unwrap();
+        let actual = rows_to_vec::<i32>(&rows);
+
+        let expected = vec![encrypted_val];
+        assert_eq!(expected, actual);
+
+        //  --------
+        // SET TENANT_2
+        let sql = format!("SET CIPHERSTASH.KEYSET_NAME = '{tenant_keyset_name_2}'");
+        let result = client.query(&sql, &[]).await;
+        assert!(result.is_ok());
+
+        // SELECT data created by TENANT_1 AS TENANT_2
+        let result = client.query(select_sql, &[&tenant_1_id]).await;
+        assert!(result.is_err());
+
+        //  --------
+        // Switch back to TENANT_1
+        let sql = format!("SET CIPHERSTASH.KEYSET_NAME = '{tenant_keyset_name_1}'");
+        let result = client.query(&sql, &[]).await;
+        assert!(result.is_ok());
+
+        // SELECT as TENANT_1
+        let result = client.query(select_sql, &[&tenant_1_id]).await;
+        assert!(result.is_ok());
+    }
+
+    ///
+    /// Tests that the `SET CIPHERSTASH.KEYSET_NAME = 'keyset_name'` command works with simple queries
+    ///
+    #[tokio::test]
+    async fn set_keyset_name_with_simple_query() {
+        trace();
+
+        clear().await;
+
+        let tenant_keyset_name_1 = std::env::var("CS_TENANT_KEYSET_NAME_1").unwrap();
+        let tenant_keyset_name_2 = std::env::var("CS_TENANT_KEYSET_NAME_2").unwrap();
+
+        // KEYSET_ID IS SCOPED TO A CONNECTION
+        // The same client/connection is used for tests
+        let client = connect_with_tls(PROXY).await;
+
+        // DEFAULT_KEYSET_ID SHOULD BE DISABLED FOR THIS TEST
+        // SET KEYSET IS REQUIRED
+        let sql = "INSERT INTO encrypted (id, encrypted_text) VALUES (1, 'hello')";
+        let result = client.simple_query(sql).await;
+        assert!(result.is_err());
+
+        // // THIS IS COUNTER INTUITIVE
+        // // Statement passes through frontend because no mapping is required
+        // // Returns without error because there is NO DATA and decrypt is never called
```

Comment:
Are these double comments left over from commenting-and-uncommenting in bulk here (and below)?